src/test/java/org/jetbrains/plugins/ideavim/action/change/insert/VisualInsertActionTest.kt

```diff
@@ -101,28 +101,76 @@ class VisualBlockInsertActionTest : VimTestCase() {
     )
   }
 
-  @TestWithoutNeovim(SkipNeovimReason.VISUAL_BLOCK_MODE)
   @Test
-  fun `test insert in non block mode`() {
-    doTest(
-      listOf("vwIHello<esc>"),
-      """
-                ${c}A Discovery
+  fun `test insert in non-block visual within single line`() {
+    val before = """
+      .
+        A ${c}Discovery
 
-                ${c}I found it in a legendary land
-                all rocks and ${c}lavender and tufted grass,
-                where it was settled on some sodden sand
-                hard by the torrent of a mountain pass.
-      """.trimIndent(),
-      """
-                Hell${c}oA Discovery
+        I ${c}found it in a legendary land
+        all rocks and lavender and tufted grass,
+        where it was settled on some sodden sand
+        hard by the torrent of a mountain pass.
+    """.trimIndent()
+    val after = """
+      .
+      Hell${c}o  A Discovery
 
-                Hell${c}oI found it in a legendary land
-                Hell${c}oall rocks and lavender and tufted grass,
-                where it was settled on some sodden sand
-                hard by the torrent of a mountain pass.
-      """.trimIndent(),
-    )
+      Hell${c}o  I found it in a legendary land
+        all rocks and lavender and tufted grass,
+        where it was settled on some sodden sand
+        hard by the torrent of a mountain pass.
+    """.trimIndent()
+    doTest(listOf($$"v$IHello<esc>"), before, after)
+    doTest(listOf("VIHello<esc>"), before, after)
+  }
+
+  @Test
+  fun `test insert in non-block visual spanning multiple lines down`() {
+    val before = """
+      .
+        A ${c}Discovery
+
+        I ${c}found it in a legendary land
+        all rocks and lavender and tufted grass,
+        where it was settled on some sodden sand
+        hard by the torrent of a mountain pass.
+    """.trimIndent()
+    val after = """
+      .
+      Hell${c}o  A Discovery
+
+      Hell${c}o  I found it in a legendary land
+        all rocks and lavender and tufted grass,
+        where it was settled on some sodden sand
+        hard by the torrent of a mountain pass.
+    """.trimIndent()
+    doTest(listOf("vjIHello<esc>"), before, after)
+    doTest(listOf("VjIHello<esc>"), before, after)
+  }
+
+  @Test
+  fun `test insert in non-block visual spanning multiple lines up`() {
+    val before = """
+      .
```

Comment:
What are these dots needed for?

Reply:
To retain the 2-space indentation on the following lines

Reply:
Please, use 'trimMargin` for it then.
https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.text/trim-margin.html